editor/src/components/navigator/navigator-item/item-label.tsx

```diff
@@ -24,154 +26,152 @@ interface ItemLabelProps {
   style?: CSSProperties
 }
 
-interface ItemLabelState {
-  name: string
-  target: NavigatorEntry
-}
+export const ItemLabel = React.memo((props: ItemLabelProps) => {
+  const elementRef = React.useRef<HTMLInputElement | null>(null)
 
-export class ItemLabel extends Component<ItemLabelProps, ItemLabelState> {
-  elementRef: HTMLInputElement | null = null
-  constructor(props: ItemLabelProps) {
-    super(props)
-    this.state = {
-      name: this.props.name,
-      target: this.props.target,
-    }
-  }
+  const [name, setName] = React.useState(props.name)
 
-  static getDerivedStateFromProps(
-    props: ItemLabelProps,
-    state: ItemLabelState,
-  ): ItemLabelState | null {
-    if (props.target === state.target || navigatorEntriesEqual(props.target, state.target)) {
-      return null
-    } else {
-      return {
-        name: props.name,
-        target: props.target,
+  const isConditionalClause = React.useMemo(() => {
+    return props.target.type === 'CONDITIONAL_CLAUSE'
+  }, [props.target])
+
+  const isActiveConditionalClause = useEditorState(
+    Substores.metadata,
+    (store) => {
+      if (!isConditionalClauseNavigatorEntry(props.target)) {
+        return false
       }
-    }
-  }
+      const parent = findMaybeConditionalExpression(
+        props.target.elementPath,
+        store.editor.jsxMetadata,
+      )
+      if (parent == null) {
+        return false
+      }
+      const activeCase = getConditionalActiveCase(
+        props.target.elementPath,
+        parent,
+        store.editor.spyMetadata,
+      )
+      if (activeCase == null) {
+        return false
+      }
+      return activeCase === props.target.clause
+    },
+    'NavigatorRowLabel isActiveBranchOfOverriddenConditional',
+  )
 
-  componentDidUpdate(prevProps: ItemLabelProps, prevState: ItemLabelState) {
-    if (!prevProps.inputVisible && this.props.inputVisible && this.elementRef != null) {
-      this.elementRef.focus()
-      this.elementRef.select()
+  React.useEffect(() => {
+    if (props.inputVisible && elementRef.current != null) {
+      elementRef.current.focus()
+      elementRef.current.select()
     }
-  }
+  }, [props.inputVisible, props.testId])
 
-  cancelRename() {
-    this.setState({
-      name: this.props.name,
-    })
-    this.props.dispatch([EditorActions.setNavigatorRenamingTarget(null)], 'leftpane')
+  const value = React.useMemo(() => {
+    return props.suffix == null ? props.name : `${props.name} ${props.suffix}`
+  }, [props.suffix, props.name])
+
+  function cancelRename() {
+    setName(props.name)
+    props.dispatch([EditorActions.setNavigatorRenamingTarget(null)], 'leftpane')
   }
 
-  renameComponent() {
-    if (isRegularNavigatorEntry(this.props.target)) {
+  function triggerRenameComponent() {
```

Comment:
this can be memoed with `React.useCallback`

Reply:
done for this one and the others in https://github.com/concrete-utopia/utopia/pull/3574/commits/a140e2ae429b349eeab6cb4402397d1d0dad147b